### PR TITLE
Add dark and high contrast theme support to ThemeProvider

### PR DESCRIPTION
**Pull request by saramarcondes**
### Testing instructions

Checkout the branch locally and pass `darkTheme`, `highContrastTheme` and `darkHighContrastTheme` to the `ThemeProvider` in `.storybook/preview.js`. Pass some different variables and ensure they're being applied correctly with the expected specificity.

### CODE REVIEW

File: packages/create-styles/src/components/ThemeProvider/ThemeProvider.js [hunk 1]
```diff
@@ -49,19 +56,48 @@ function ThemeProvider({
...
+	darkHighContrastTheme = {},
 	...props
 }) {
 	/**
 	 * Hydrates global styles (via injectGlobal). This is necessary as there may
 	 * be a chance that <ThemeProvider /> renders before any other (styled)
 	 * component. Injecting global styles early in this manner allows for
 	 * the initial render of theme styles (which also uses injectGlobal)
-	 * to be sequences correctly.
+	 * to be sequenced correctly.
```
Comment (ItsJonQ) [inline packages/create-styles/src/components/ThemeProvider/ThemeProvider.js:69]: Thank you 🙏

**Review by ItsJonQ — APPROVED** [review]
@saramarcondes 🚀 from me! Tested it in Storybook, and the new theme modes are working as expected, including local settings (`isGlobal={false}`) and global settings.

(Not a blocker) The thing I'm not sure of is it looks like the `themeStyles` object is newly created every render. Previously, it was the instance provided by the substate store (powered by `zustand`).

It doesn't appear to be a problem. But something to keep an eye on.

Lemme know if you have any thoughts. If you think it's 👍 , then I'll merge this up!

**Comment by saramarcondes** [comment]
@ItsJonQ 

> The thing I'm not sure of is it looks like the `themeStyles` object is newly created every render

Is this a new thing from my changes? I'm not really sure I would know where to start to try to address this... do you have any pointers?

Also... have you observed a performance deficit due to this?

**Comment by ItsJonQ** [comment]
> Is this a new thing from my changes?

@saramarcondes I believe so :)

>  I'm not really sure I would know where to start to try to address this...

The theme object was created/managed with this:

https://github.com/ItsJonQ/g2/blob/a5f1fb1a74f78c72b0678e3e5ce919c367968960/packages/create-styles/src/components/ThemeProvider/ThemeProvider.utils.js#L141

That `theme` object is only changed when the values are updated. Otherwise, it should be the same object being returned.

With this update, we're getting the 4 `theme` objects and merging them on every render.

> Also... have you observed a performance deficit due to this?

Yes and no 😂 .

Yes, if you needed to update/re-render the `ThemeProvider` often (e.g. changing the `colorAdmin` variable using a color picker).

However! There's an interesting thing about G2's ThemeProvider. As you've discovered, because of how G2 handles theming/variables, it doesn't require to you to put content (`children`) inside of `ThemeProvider`. Which means, there's no gigantic React node-tree walk down/up per render.

This only applies to "global" updates, not scoped theme updates (controlled by the `isGlobal` prop).

☝️ With this being said... we may be okay for now :).

Lemme know what you think. If this makes sense, then we can merge it up

File: .storybook/themes/blueberry.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { createTheme } from '@wp-g2/styles';
 
-const theme = createTheme(({ get, space, ...theme }) => {
```
Comment (ItsJonQ) [inline .storybook/themes/blueberry.js:3]: Thank you for fixing this 🙈

**Pull request merged**